app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java

```diff
@@ -411,6 +441,54 @@ public void ensureHabitAdditionFails_dateEmpty() {
         deleteTestHabit(testHabit);
     }
 
+    @Test
+    public void ensureHabitAdditionFails_NoDays(){
+
+        // Asserts that the current activity is the MainActivity. Otherwise, show “Wrong Activity”
+        _solo.assertCurrentActivity(WRONG_ACTIVITY_MESSAGE, MainActivity.class);
+
+        // click on the Habit List tab in the bottom navigation bar
+        _solo.clickOnView(_solo.getView(R.id.navigation_dashboard));
+
+        // click on add habit floating action button to add habit
+        _solo.clickOnView(_solo.getView(R.id.add_habit_fab));
+
+        // Create test habit
+        Habit testHabit = new Habit("addHabitNoDays", "Test Reason", new Date(), "MO WE FR", HABIT_ID, EMPTY_HABIT_EVENT_LIST);
+
+        // Enter title
+        setFieldInAddHabitDialogBox(HABIT_TITLE_FIELD, testHabit.getTitle());
+
+        // Enter reason
+        setFieldInAddHabitDialogBox(HABIT_REASON_FIELD, testHabit.getReason());
+
+        // Enter date
+        enterCurrentDateInAddHabitDialogBox();
+
+        // Click confirm
+        clickConfirmButtonInAddHabitDialogBox();
+
+        // Check for toast message
+        checkForToastMessage(NO_DAYS_SELECTED_ERROR_MESSAGE);
+        //assertTrue(isTextOnScreen(NO_DAYS_SELECTED_ERROR_MESSAGE));
+
+        // Select days
+        setDaysInAddHabitDialogBox();
+
+        // Click confirm
+        clickConfirmButtonInAddHabitDialogBox();
+
+
+        // Check that the current fragment is the habit list
+        assertTextOnScreen(HABIT_LIST_TEXT);
+
+        // Ensure added Habit is present in the list
+        assertTextOnScreen(testHabit.getTitle());
+
+        deleteTestHabit(testHabit);
+
+    }
+
```

Comment:
This test ensures that the error message of having no days selected pops up. Currently, I have the same problem as @JuliePilz where the toast checks don't pop up, but I have manually ran the test and it works.